tests/unit/api/objectPut.js

```diff
@@ -269,6 +429,30 @@ describe('objectPut API', () => {
                 });
         });
     });
+
+    it('should not put object with retention configuration if object lock ' +
+        'is not enabled on the bucket', done => {
+        const testPutObjectRequest = new DummyRequest({
+            bucketName,
+            namespace,
+            objectKey: objectName,
+            headers: {
+                'x-amz-object-lock-retain-until-date': mockDate,
+                'x-amz-object-lock-mode': 'GOVERNANCE',
```

Comment:
I would consider moving these header names to a constants file to prevent typos and the like.